qutebrowser/completion/completionwidget.py

```diff
@@ -255,3 +257,26 @@ def showEvent(self, e):
         if scrollbar is not None:
             scrollbar.setValue(scrollbar.minimum())
         super().showEvent(e)
+
+    def hide(self):
+        super().hide()
+
+        if config.get('completion', 'auto-open') == True:
+            return
+
+        cmd = objreg.get('status-command', scope='window', window=self._win_id)
+        try:
+            cmd.cursorPositionChanged.disconnect(cmd.update_completion)
+            cmd.textChanged.disconnect(cmd.update_completion)
+        # Don't fail if not connected
+        except TypeError:
+            pass
+
+    def _open_completion_if_needed(self):
+        if config.get('completion', 'auto-open') == False:
+            update_completion = pyqtSignal()
```

Comment:
What's this supposed to do? I don't think there's any valid use of `pyqtSignal()` outside of a class body, and here that's actually a signal on `cmd`. Actually I'd be surprised if it doesn't raise an exception, but I haven't tried yet.